src/ray/core_worker/gcs_server_address_updater.cc

```diff
@@ -64,8 +64,23 @@ void GcsServerAddressUpdater::UpdateGcsServerAddress() {
              "the worker might crash. Connection status "
           << status;
       if (failed_ping_count_ == RayConfig::instance().ping_gcs_rpc_server_max_retries()) {
-        RAY_LOG(FATAL) << "Failed to receive the GCS address from the raylet for "
-                       << failed_ping_count_ << " times. Killing itself.";
+        std::stringstream os;
+        os << "Failed to receive the GCS address for " << failed_ping_count_
+           << " times without success. The worker will exit ungracefully. It is because ";
+        if (IsRayletFailed(RayConfig::instance().RAYLET_PID())) {
+          RAY_LOG(WARNING) << os.str()
```

Comment:
@scv119 actually, we should think about how to handle the case where it is a driver (e.g., head node raylet is failed). I think if it is a driver, it's always ideal to print this sort of logs to ERROR severity because otherwise driver will fail silently.

Reply:
do we have document of RAY_LOG? i'm always confused what log is streamed to driver.

Reply:
We don't. Rule of thumb is ERROR level logs (which is streamed to stderr) are always streamed to the driver.